### PR TITLE
[COST-3370] - drop ranking field from query data

### PR DESCRIPTION
**Pull request by maskarb**
## Jira Ticket

[COST-3370](https://issues.redhat.com/browse/COST-3370)

## Description

This change will ensure the ranking field is dropped from the original query data.
This query:
```
/?filter[limit]=50&filter[offset]=0&filter[resolution]=monthly&filter[time_scope_units]=month&filter[time_scope_value]=-2&group_by[project]=*&order_by[request]=desc
```
would result in the  `ranks_by_day` and `data_frame` containing a `request` column. When these dataframes were merged, it resulted in a new DF with `request_x` and `request_y` fields. Further along in the query_handler, when ordering the data, we would try to sort by `request` but this field no longer existed in the data.

## Testing

1. Ingest OCP data.
2. Go to http://localhost:8000/api/cost-management/v1/reports/openshift/compute/?filter[limit]=50&filter[offset]=0&filter[resolution]=monthly&filter[time_scope_units]=month&filter[time_scope_value]=-2&group_by[project]=*&order_by[request]=desc
3. see data.
4. Go to http://localhost:8000/api/cost-management/v1/reports/openshift/compute/?filter[limit]=50&filter[offset]=0&filter[resolution]=monthly&filter[time_scope_units]=month&filter[time_scope_value]=-2&group_by[project]=*&order_by[delta]=desc&delta=request__limit
5. see data.

## Notes

...

### CODE REVIEW

**Review by adberglund — DISMISSED** [review]

File: koku/api/report/queries.py [hunk 1]
```diff
@@ -1021,9 +1021,9 @@ def _group_by_ranks(self, query, data):  # noqa: C901
             if rank_value not in rankings:
                 rankings.append(rank_value)
                 distinct_ranks.append(rank)
-        return self._ranked_list(data, distinct_ranks)
+        return self._ranked_list(data, distinct_ranks, list(rank_annotations.keys()))
```
Comment (samdoran) [inline koku/api/report/queries.py:1024]: Passing a dictionary into `list()` will return a list of keys.
```suggestion
        return self._ranked_list(data, distinct_ranks, list(rank_annotations))
```

File: koku/api/report/queries.py [hunk 2]
```diff
@@ -1021,9 +1021,9 @@ def _group_by_ranks(self, query, data):  # noqa: C901
...
                 rankings.append(rank_value)
                 distinct_ranks.append(rank)
-        return self._ranked_list(data, distinct_ranks)
+        return self._ranked_list(data, distinct_ranks, list(rank_annotations))
 
-    def _ranked_list(self, data_list, ranks):
+    def _ranked_list(self, data_list, ranks, rank_field=None):
         """Get list of ranked items less than top.
 
         Args:
```
Comment (samdoran) [inline koku/api/report/queries.py:1029]: New arg needs to be added to the doc string.

File: koku/api/report/queries.py [hunk 3]
```diff
@@ -1075,6 +1077,10 @@ def _ranked_list(self, data_list, ranks):
         # Cross join ranks and days to get each field/rank for every day in th query
         ranks_by_day = rank_data_frame.merge(day_data_frame, how="cross")
 
+        # add the ranking column if it still exists in both dataframes
+        if rank_field and rank_field[0] in ranks_by_day.columns.intersection(data_frame.columns):
```
Comment (samdoran) [inline koku/api/report/queries.py:1081]: Is `rank_field` expected to be a single item list passed into this method?

Comment (maskarb) [reply koku/api/report/queries.py:1081]: yesf

Comment (maskarb) [reply koku/api/report/queries.py:1081]: On second look, it does appear that rank_field may contain more than 1 entry. I'll update this logic here to check for all possible entries.

Comment (maskarb) [reply koku/api/report/queries.py]: Here is a query which results in a `rank_field` containing multiple fields:
```
api/cost-management/v1/reports/openshift/compute/?filter[limit]=50&filter[offset]=0&filter[resolution]=monthly&filter[time_scope_units]=month&filter[time_scope_value]=-2&group_by[project]=*&order_by[delta]=desc&delta=request__limit
```

File: koku/api/report/queries.py [hunk 4]
```diff
@@ -1075,6 +1078,11 @@ def _ranked_list(self, data_list, ranks):
         # Cross join ranks and days to get each field/rank for every day in th query
         ranks_by_day = rank_data_frame.merge(day_data_frame, how="cross")
 
+        # add the ranking columns if they still exists in both dataframes
+        rank_fields.intersection_update(set(ranks_by_day.columns.intersection(data_frame.columns)))
```
Comment (samdoran) [inline koku/api/report/queries.py:1082]: `set` methods accept an iterable and don't require a `set`. The `pandas.core.indexes.numeric.Int64Index` returned by the `intersection` method is iterable.
```suggestion
        rank_fields.intersection_update(ranks_by_day.columns.intersection(data_frame.columns))
```

**Review by adberglund — APPROVED** [review]

**Pull request merged**